File: Latihan2-py/Hardware.py
from Product import Product # Superclass

class Hardware(Product):
    # Constructor
    def __init__(self):
        self.__brand = ""
        self.__model = ""
    
    # Get-set Brand
    def setBrand(self, brand):
        self.__brand = brand

    def getBrand(self):
        return self.__brand
    
    # Get-set Model
    def setModel(self, model):
        self.__model = model

    def getModel(self):
        return self.__model

File: Latihan2-cpp/Hardware.cpp
#include "Product.cpp"

using namespace std;

class Hardware : public Product{

    private:

    string brand;
    string model;

    public:

    // Constructor
    Hardware(){
    }

    Hardware(string brand, string model){
        this->brand = brand;
        this->model = model;
    }

    // Get-set Brand
    void setBrand(string brand){
        this->brand = brand;
    }

    string getBrand(){
        return this->brand;
    }

    // Get-set Model
    void setModel(string model){
        this->model = model;
    }

    string getModel(){
        return this->model;
    }

    // Deconstructor
    ~Hardware(){
    }

};